src/components/AnimateSharedLayout/stack.ts

```diff
@@ -210,17 +205,29 @@ export class LayoutStack {
         return this.lead && this.lead?.presence !== Presence.Exiting
     }
 
-    shouldStackAnimate() {
+    shouldStackAnimate(): boolean {
+        if (this.lead && this.lead?.isPresent)
+            return this.shouldPrimaryAnimate(this.lead, this.follow)
+
+        if (this.follow)
+            return this.shouldPrimaryAnimate(this.follow, this.lead)
+
         return true
-        // return this.lead && this.lead?.isPresent
-        //     ? this.lead?.props?._shouldAnimate === true
-        //     : this.follow && this.follow?.props._shouldAnimate === true
+    }
+
+    shouldPrimaryAnimate(
+        primary: HTMLVisualElement,
+        secondary?: HTMLVisualElement
+    ) {
+        if (primary.config._shouldAnimateLayout === undefined) return true
+        secondary && secondary.config._shouldAnimateLayout?.(false)
+        return primary.config._shouldAnimateLayout(true) as boolean
```

Comment:
I'd love thoughts on a better way to do, or at very least name this. 

Essentially we need to call a function defined on the follow to ensure that the framer shouldAnimate context has the follows `props` assigned to the layoutId. That way when `_shouldAnimateLayout()` is called on the lead, we can compare props and return a boolean.

Framer side, we're creating a callback like `(boolean) => boolean ? calculate() : register()`.

So we could do something like _prepareShouldAnimate(): void and remove the need to pass a boolean, but my gut feeling is to keep the surface area of this framer-specific stuff as small as possible, hence the single prop 🤔